playbooks/tasks/portal-accounts-generate-jwks.json.yml

```diff
@@ -0,0 +1,47 @@
+---
+# Generate Accounts jwks.json config file according to skynet-accounts README.
+# Sometimes the generated jwks.json file is empty so we retry until we
+# successfully generate the content.
+
+- name: Increment jwks retry count
+  set_fact:
+    jwks_retries_count: "{{ jwks_retries_count | int + 1 }}"
+
+- name: Generate JWKS config according to README in skynet-accounts repo
+  local_action:
+    module: community.docker.docker_container
+    name: jwks-config-generator
+    image: alpine:3.16.0
+    volumes:
+      # Following host:guest dirs must be the same because our container
+      # call golang container (docker in docker) and we need to mount the
+      # same working directory so we get generated files to the `output`
+      # directory
+      - "{{ jwks_repo_dir }}:{{ jwks_repo_dir }}"
+      # We call another docker container from our container (docker in
+      # docker) so we need to pass docker socket
+      - /var/run/docker.sock:/var/run/docker.sock
+    command:
+      - "sh -c"
+      # String of commands is required to be enclosed in single quotes. We:
+      # - install docker
+      # - install make
+      # - cd to skynet-accounts repo
+      # - generate jwks config
+      - "'apk add docker && apk add make && cd {{ jwks_repo_dir }} && make docker-generate'"
+    container_default_behavior: no_defaults
+    auto_remove: True
+    detach: False
+
+- name: Read generated jwks.json
+  set_fact:
+    accounts_jwks_data: "{{ lookup('file', jwks_repo_dir + '/output/jwks.json') }}"
+
+- name: Fail if failed to generate jwks.json file several times
+  fail:
+    msg: "Ansible was not able to generate jwks.json file for {{ jwks_retries_max }} times"
+  when: accounts_jwks_data == '' and jwks_retries_count | int == jwks_retries_max
+
+- name: Retry if generated jwks.json file is empty
+  include_tasks: tasks/portal-accounts-generate-jwks.json.yml
+  when: accounts_jwks_data == ''
```

Comment:
why is that even a possibility ? maybe "make docker-generate" doesn't have enough time to write to the file or volumes did not sync yet before the container is killed, can you try adding a 1 second sleep after "make docker-generate" ?

Reply:
When running locally (manually) according to the Accounts readme, in aprox. 80% cases the output was generated, in 20% it was missing. I have created Accounts PR to prolong the sleep.

When running via Ansible (which runs inside docker container) it a) took longer to run, b) syncing the files on volumes took sometimes much longer.

So the safest way is to retry.
Otherwise the playbook fails for the portal operator and he/she needs to rerun the complete setup playbook.

Reply:
Not an issue anymore, using `oathkeeper` image now.